### PR TITLE
fix(onboard): guide reviewed defaults and auto-model recovery

### LINKED ISSUE #214: [Bug] Harden onboarding exit and explicit provider defaults
## Summary

Interactive onboarding on `alpha-test` still had two first-run regressions in alpha testing:

- choice/input screens did not tell users how to cancel cleanly before config write
- MiniMax onboarding could fail on model discovery (`401`) because setup depended too heavily on live model catalog probing while `model = auto`

The deeper issue is architectural: hidden provider-owned runtime fallback models are the wrong fix. They create invisible behavior and ongoing provider drift risk.

## Expected behavior

- Every onboarding screen explains a clean cancel path before config write.
- Providers with a reviewed onboarding default model can complete setup with an explicit model even when model catalog discovery is unavailable during setup.
- Runtime, `onboard`, and `doctor` only advertise fallback continuation when the operator explicitly configured `preferred_models`.
- Quick-start docs include a short WSL note for operators testing in WSL.

## Reproduction steps

1. Check out `origin/alpha-test` at `4e87e6b`.
2. Run `loongclaw onboard` in a CLI environment.
3. Select `minimax` and keep the default `model = auto`.
4. Configure credentials and continue into preflight.
5. Observe onboarding surfaces a provider model probe failure instead of steering the user toward an explicit reviewed model.
6. Try to back out from choice/input screens and note there is no explicit `Esc` cancellation hint.

## Correct fix direction

- keep the `Esc` cancellation improvement
- use onboarding-only explicit provider defaults for reviewed providers such as MiniMax
- do **not** seed hidden provider-owned runtime `preferred_models`
- keep `preferred_models` as an explicit operator-configured fallback path only

## Version / commit

`alpha-test@4e87e6b`

## Contribution track impact

Track A (routine/low-risk)

### PR DESCRIPTION
**Pull request by chumyin**
## Summary

- What changed?
  - Added shared `ModelCatalogProbeRecovery` so provider config, onboarding, and doctor classify model-catalog probe failures from the same recovery source.
  - Added DeepSeek to the reviewed onboarding defaults with `deepseek-chat`, alongside MiniMax `MiniMax-M2.5`, so reviewed providers can land on an explicit model even when setup-time catalog probing is unavailable.
  - Onboarding and doctor now turn `model = auto` probe failures into actionable remediation: rerun onboarding and accept the reviewed model when available, or set `provider.model` / `preferred_models` explicitly.
  - `doctor` next steps and non-interactive onboarding failures now surface the correct recovery path instead of only pointing operators at `--skip-model-probe`.
  - Updated README, `README.zh-CN`, and onboarding product specs to document reviewed defaults and explicit auto-model recovery guidance.
- Why this change is needed?
  - Alpha-test feedback exposed MiniMax `401` failures, but the real bug was broader: any provider left on `model = auto` without explicit recovery candidates would fail catalog probing without giving the operator a clear next step.
  - Official provider docs and OpenClaw’s onboarding pattern both favor explicit first-run model selection or manual model entry over hidden runtime guesses.
  - This keeps `preferred_models` operator-owned, reduces provider drift, and makes first-run recovery consistent across reviewed providers instead of shipping another provider-specific patch.

## Scope

- [x] Small and focused
- [x] Includes docs updates (if needed)
- [x] No unrelated refactors

## Risk Track

- [x] Track A (routine/low-risk)
- [ ] Track B (higher-risk/policy-impacting)

If Track B, include design/risk notes:

## Validation

- [x] `cargo fmt --all -- --check`
- [x] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [x] `cargo test --workspace --all-features`
- [x] Additional scenario/benchmark checks (if applicable)
- [x] If this changes config/env fallback, limits, or defaults: include before/after behavior and regression coverage for explicit path, fallback path, and boundary values
- [x] If tests mutate process-global env: document how state is restored or serialized

Additional notes:
- Fresh local validation on head `3aeb941` included:
  - `cargo test -p loongclaw-app provider::tests:: -- --nocapture`
  - `cargo test -p loongclaw-daemon onboard_cli::tests:: -- --nocapture`
  - `cargo test -p loongclaw-daemon doctor_cli::tests:: -- --nocapture`
  - `cargo test --workspace --all-features`
- The current head is rebased onto the existing PR branch follow-up commit `f696a19`, so the earlier onboarding exit-guidance/prompt-wording fixes remain intact.
- Before: DeepSeek still behaved like a generic `model = auto` provider during probe failure, and `onboard`, `doctor`, and doctor next steps could drift or over-suggest `--skip-model-probe`.
- After: reviewed providers surface explicit onboarding defaults, and `model = auto` failures now branch into reviewed-onboarding recovery vs manual `provider.model` / `preferred_models` recovery from a shared classification.
- Regression coverage includes provider config defaults, onboarding model selection defaults, onboarding/doctor failure messaging, doctor next-step guidance, and non-interactive preflight error surfacing.
- Env-mutating onboarding tests continue to serialize and restore provider/channel environment state via `DetectedEnvironmentGuard`.
- Independent review on the final diff reported no findings.

## Linked Issues

Closes #214

## Summary by CodeRabbit

* **New Features**
  * Cancel onboarding with Esc (then Enter) before config is written
  * Onboarding pre-fills reviewed recommended models for MiniMax and DeepSeek
  * Onboarding accepts configured preferred-model fallbacks as a non-interactive remediation

* **Bug Fixes / UX**
  * Clearer actionable remediation when model discovery fails (rerun onboarding or set explicit model/preferred_models)

* **Documentation**
  * Expanded onboarding docs and WSL/systemd guidance; added onboarding design/plan notes

### CODE REVIEW

File: crates/daemon/src/onboard_cli.rs [hunk 1]
```diff
@@ -5051,24 +5132,36 @@ mod tests {
...
 
         fn prompt_required(&mut self, _label: &str) -> CliResult<String> {
-            self.inputs
+            let value = self
+                .inputs
                 .pop_front()
-                .ok_or_else(|| "missing required test input".to_owned())
+                .ok_or_else(|| "missing required test input".to_owned())?;
+            ensure_onboard_input_not_cancelled(value)
         }
```
[the remark this answers is not in the document]
Comment (chumyin) [reply crates/daemon/src/onboard_cli.rs:5206]: Fixed in ec1e5f2. `TestOnboardUi` now mirrors `StdioOnboardUi`: cancel detection only applies to actual queued input, `prompt_required` trims like stdio, and regression tests pin both behaviors.

File: docs/plans/2026-03-16-provider-onboarding-default-model-implementation-plan.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+onboarding flow to use it when `model` is still `auto`, and remove built-in
+runtime fallback seeding from `preferred_models`. Runtime, doctor, and onboarding
+should only advertise fallback continuation when the operator explicitly
+configured `preferred_models`.
+
+**Tech Stack:** Rust, cargo test, cargo clippy
+
+---
+
+### Task 1: Add failing provider-config tests
```
[the remark this answers is not in the document]
Comment (chumyin) [reply docs/plans/2026-03-16-provider-onboarding-default-model-implementation-plan.md]: Fixed in ec1e5f2. The task headings in `docs/plans/2026-03-16-provider-onboarding-default-model-implementation-plan.md` now use H2 so the document hierarchy is H1 -> H2.

**Comment by chumyin** [comment]
Tracked the next provider-wide follow-up separately in #236 so this PR can stay narrowly focused on the validated alpha-test fixes: reviewed onboarding defaults, actionable auto-model recovery, and clearer first-run exit guidance.

**Comment by gh-xj** [comment]
LoongClaw AI scientist has claimed this case and is running review execution.
